src/py/flwr/server/utils/validator_test.py

```diff
@@ -0,0 +1,118 @@
+# limitations under the License.
+# ==============================================================================
+"""Validator tests."""
+import unittest
+from typing import List, Tuple
+
+from flwr.proto.node_pb2 import Node
+from flwr.proto.task_pb2 import Task, TaskIns, TaskRes
+from flwr.proto.transport_pb2 import ClientMessage, ServerMessage
+
+from .validator import validate_task_ins_or_res
+
+
+class ValidatorTest(unittest.TestCase):
+    """Test validation code in state."""
+
+    def test_task_ins(self) -> None:
+        """Test is_valid task_ins."""
+        # Prepare
+        # (consumer_node_id, anonymous)
+        valid_ins = [(0, True), (1, False)]
+        invalid_ins = [(0, False), (1, True)]
+
+        # Execute & Assert
+        for consumer_node_id, anonymous in valid_ins:
+            msg = create_task_ins(consumer_node_id, anonymous)
+            val_errors = validate_task_ins_or_res(msg)
+            self.assertFalse(val_errors)
+
+        for consumer_node_id, anonymous in invalid_ins:
+            msg = create_task_ins(consumer_node_id, anonymous)
+            val_errors = validate_task_ins_or_res(msg)
+            self.assertTrue(val_errors)
+
+    def test_is_valid_task_res(self) -> None:
+        """Test is_valid task_res."""
+        # Prepare
+        # (consumer_node_id, anonymous, ancestry)
+        valid_res: List[Tuple[int, bool, List[str]]] = [
+            (0, True, ["1"]),
+            (1, False, ["1"]),
+        ]
+
+        invalid_res: List[Tuple[int, bool, List[str]]] = [
+            (0, False, []),
+            (0, False, ["1"]),
+            (0, True, []),
+            (1, False, []),
+            (1, True, []),
+            (1, True, ["1"]),
+        ]
+
+        # Execute & Assert
+        for consumer_node_id, anonymous, ancestry in valid_res:
+            msg = create_task_res(consumer_node_id, anonymous, ancestry)
+            val_errors = validate_task_ins_or_res(msg)
+            self.assertFalse(val_errors)
+
+        for consumer_node_id, anonymous, ancestry in invalid_res:
+            msg = create_task_res(consumer_node_id, anonymous, ancestry)
+            val_errors = validate_task_ins_or_res(msg)
+            self.assertTrue(val_errors)
+
+
+def create_task_ins(
+    consumer_node_id: int, anonymous: bool, delivered_at: str = ""
+) -> TaskIns:
+    """Create a TaskIns for testing."""
+    consumer = Node(
+        node_id=consumer_node_id,
+        anonymous=anonymous,
+    )
+    task = TaskIns(
+        task_id="",
+        group_id="",
+        workload_id="",
+        task=Task(
+            delivered_at=delivered_at,
+            producer=Node(node_id=0, anonymous=True),
+            consumer=consumer,
+            legacy_server_message=ServerMessage(
+                reconnect_ins=ServerMessage.ReconnectIns()
+            ),
+        ),
+    )
+    return task
+
+
+def create_task_res(
+    consumer_node_id: int, anonymous: bool, ancestry: List[str]
```

Comment:
Should this be `producer_node_id`? For now, the `consumer_node_id` is always expected to be `0`

Reply:
Fixing it.